GitUI/BranchTreePanel/RepoObjectsTree.Nodes.cs

```diff
@@ -366,7 +367,16 @@ static void EnsureNodeVisible(IntPtr hwnd, TreeNode node)
         private abstract class Node : INode
         {
             public readonly Nodes Nodes;
-            protected Tree Tree => Nodes.Tree;
+
+            protected Tree Tree
+            {
+                get
+                {
+                    Assumes.NotNull(Nodes.Tree);
+                    return Nodes.Tree;
+                }
+            }
```

Comment:
One usage needs to set a null `Tree` property, however we never want to allow reading a null from here. Changed to allow null in the ctor but throw if it's read.